src/cnab.rs

```diff
@@ -20,39 +20,47 @@ pub struct Bundle {
     ///
     /// 'install', 'upgrade', and 'uninstall' are default actions, but additional actions
     /// may be defined here.
-    pub actions: Option<HashMap<String, Action>>,
+    pub actions: Option<BTreeMap<String, Action>>,
     /// The list of configurable credentials.
     ///
     /// Credentials are injected into the bundle's invocation image at startup time.
-    pub credentials: Option<HashMap<String, Credential>>,
+    pub credentials: Option<BTreeMap<String, Credential>>,
     /// This field allows for additional data to described in the bundle.
     ///
     /// This data should be stored in key/value pairs, where the value is undefined by
     /// the specification (but must be representable as JSON).
-    pub custom: Option<HashMap<String, serde_json::Value>>,
+    pub custom: Option<BTreeMap<String, serde_json::Value>>,
+
+    /// The JSON Schemata describing the parameters
+    ///
+    /// TODO: Should use a suitable Rust library as the target for this.
+    pub definitions: Option<BTreeMap<String, serde_json::Value>>,
+
     /// description is a short description of this bundle
     pub description: Option<String>,
     /// The list of images that comprise this bundle.
     ///
     /// Each image here is considered a constituent of the application described by this
     /// bundle.
-    pub images: Option<HashMap<String, Image>>,
-    /// inovcation_images is the list of available bootstrapping images for this bundle
+    pub images: Option<BTreeMap<String, Image>>,
```

Comment:
Just for my own understanding, was `BTreeMap` preferred over `HashMap` for predictable ordering or some other reason?

Reply:
Predictable ordering.